### PR TITLE
More infestation ports

### PR DESCRIPTION
**Pull request by EgorDinamit**
## About the Pull Request

Mostly a port of vlggms/tegustation-bay12/pull/523
Reworks structures, adds pike burrow and much more.

## Why It's Good For The Game

It's now better, read original PR for more info.

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/infestation/structures/_infestation_structure.dm [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/obj/infestation_structure
```
Comment (aidan-bowman) [inline code/modules/mob/living/simple_animal/hostile/infestation/structures/_infestation_structure.dm:1]: add `abstract_type = /obj/infestation_structure` here

File: code/modules/mob/living/simple_animal/hostile/infestation/aggregate.dm [hunk 2]
```diff
@@ -95,4 +100,5 @@
 	var/mob/living/simple_animal/hostile/infestation/meatchip/M = new(get_turf(src))
 	if(!throw_target)
 		throw_target = pick(getcircle(get_turf(src), 3))
-	M.throw_at(get_turf(throw_target), 3, 6)
+	M.throw_at(get_turf(throw_target), 3, 1, src)
+	addtimer(CALLBACK(M, /mob/living/simple_animal/hostile/infestation/meatchip/proc/TimedDeath), 15 SECONDS)
```
Comment (aidan-bowman) [inline code/modules/mob/living/simple_animal/hostile/infestation/aggregate.dm]: this is not 515 compliant

File: code/modules/mob/living/simple_animal/hostile/infestation/lurker.dm [hunk 3]
```diff
@@ -0,0 +1,82 @@
+// High ranged damage, Medium health, Low mobility.
+// Recommended strategy is to rush it, as its melee damage is subpar and has no armor penetration.
+/mob/living/simple_animal/hostile/infestation/lurker
+	name = "lurker"
+	desc = "A spider-like creature with multiple spikes protruding from its \"legs\"."
+	icon = 'icons/mob/simple_animal/abominable_infestation/48x48.dmi'
+	// !!!! I am not a very good spriter, please update it later !!!!
```
Comment (aidan-bowman) [inline code/modules/mob/living/simple_animal/hostile/infestation/lurker.dm:7]:
```suggestion
	// TODO: improve these sprites
```

or better yet make an issue (if this is merged)

**Comment by aidan-bowman** [comment]
egor try to follow code standards challenge (impossible)

File: code/modules/mob/living/simple_animal/hostile/infestation/lurker.dm (hunk 3 above)
Comment (EgorDinamit) [reply code/modules/mob/living/simple_animal/hostile/infestation/lurker.dm:7]: Lurker is not available normally so this is fine for now.

**Review by Tsurupeta — APPROVED** [review]

File: code/modules/mob/living/simple_animal/hostile/infestation/structures/_infestation_structure.dm [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	name = "infestation structure base"
+	desc = "You are not meant to see this."
+
+	density = TRUE
+	opacity = 1
+	anchored = TRUE
+	mouse_opacity = 2
+	abstract_type = /obj/infestation_structure
```
Comment (Tsurupeta) [inline code/modules/mob/living/simple_animal/hostile/infestation/structures/_infestation_structure.dm]: move it to under type definition. :3

**Review by Tsurupeta — DISMISSED** [review]

**Review by Tsurupeta — APPROVED** [review]

**Pull request merged**